youtube_dl/extractor/spreaker.py

```diff
@@ -0,0 +1,192 @@
+            new_items = playlist_.get('items', [])
+            if not new_items:
+                break
+            items += new_items
+            next_url = playlist_.get('next_url')
+            page_no += 1
+
+        urls = [x['api_url'] for x in items]
+        ret = []
+        for index, url in enumerate(urls):
+            data = self._download_json(url, None)['response']['episode']
+            dict_ = SpreakerIE._spreaker_episode_data_to_info(data)
+            dict_.update({
+                'playlist_id': compat_str(data['show_id']),
+                'playlist_title': data['show']['title'],
+                'playlist_index': index,
+            })
+            ret.append(dict_)
+
+        return self.playlist_result(ret,
+                                    data['show_id'],
+                                    data['show']['title'])
+
+
+class SpreakerAPIEpisodeIE(InfoExtractor):
+    IE_NAME = 'spreaker:api'
+    _VALID_URL = r'^https?://(?:api\.)?spreaker\.com/(?:download/)?episode/(?P<id>[0-9]+)(?:/[^\.]+\.mp3$)?'
+    _TESTS = [
+        {
+            'url': 'https://api.spreaker.com/episode/12534508',
+            'info_dict': {
+                'id': '12534508',
+                'ext': 'mp3',
+                'title': 'Marketing Your Music - Part 2',
+                'upload_date': '20170809',
+                'uploader': 'SWM',
+                'uploader_id': 9780658,
+            },
+        },
+        {
+            'url': 'https://api.spreaker.com/download/episode/12534508/swm_ep15_how_to_market_your_music_part_2.mp3',
+            'info_dict': {
+                'id': '12534508',
+                'ext': 'mp3',
+                'title': 'Marketing Your Music - Part 2',
+                'upload_date': '20170809',
+                'uploader': 'SWM',
+                'uploader_id': 9780658,
+            },
+        },
+    ]
+
+    def _real_extract(self, url):
+        episode_id = self._match_id(url)
+        if not re.match(r'^[0-9]+$', episode_id):
+            raise ExtractorError('Invalid ID')
+
+        url = 'https://api.spreaker.com/episode/%s' % (episode_id,)
+        data = self._download_json(url, episode_id)['response']['episode']
+        if not data['download_enabled']:
+            raise ExtractorError('Not supported yet')
+
+        return SpreakerIE._spreaker_episode_data_to_info(data)
+
+
+class SpreakerIE(InfoExtractor):
+    IE_NAME = 'spreaker'
+    _VALID_URL = r'^https?://(?:www\.)?spreaker\.com/user/[a-z0-9_-]+/[a-z0-9_-]'
+    _TEST = {
+        'url': 'https://www.spreaker.com/user/9780658/swm-ep15-how-to-market-your-music-part-2',
+        'info_dict': {
+            'id': '12534508',
+            'ext': 'mp3',
+            'title': 'Marketing Your Music - Part 2',
+            'upload_date': '20170809',
+            'uploader': 'SWM',
+            'uploader_id': 9780658,
+        },
+    }
+
+    @staticmethod
+    def _spreaker_episode_data_to_info(data):
```

Comment:
This is supposed to be in a class that does extraction, not in a class that delegates.